ql/experimental/credit/syntheticcdo.cpp

```diff
@@ -107,11 +107,14 @@ namespace QuantLib {
     Real SyntheticCDO::protectionLegNPV() const {
         calculate();
         if(side_ == Protection::Buyer) return -protectionValue_;
-        return premiumValue_;
+        return protectionValue_;
     }
 
     Rate SyntheticCDO::fairPremium () const {
         calculate();
+        if(premiumValue_ == 0) {
```

Comment:
This should be checked using a `QL_REQUIRE` macro instead (so it can optionally add file/line information).

Reply:
Thanks, have updated.